coalib/testing/LocalBearTestHelper.py

```diff
@@ -185,10 +201,11 @@ def check_results(self,
                                   create_tempfile=create_tempfile,
                                   tempfile_kwargs=tempfile_kwargs,
                                   settings=settings)
+        self.addTypeEqualityFunc(list, self.assertObjectsEqual)
```

Comment:
Ouch haven't thought about that we pass results with lists... here we could just iterate over all passed results and compare them one by one with `assertEqual`. And we still register `addTypeEqualityFunc` with `Result` globally (or an ABC-type of classes using `generate_eq` and `generate_ordering` which would be the ideal solution).

Hm this is better than nothing, but ideally we would be able to have nicer outputs for collections... don't know something yet, maybe you also get some ideas how to make `self.assertObjectsEqual` working automatically everywhere in our tests, even those ones not testing only bears.